sound/soc/sof/topology.c

```diff
@@ -3474,12 +3481,19 @@ EXPORT_SYMBOL(snd_sof_init_topology);
 
 int snd_sof_load_topology(struct snd_sof_dev *sdev, const char *file)
 {
+	struct firmware vfe_fw;
 	const struct firmware *fw;
 	int ret;
 
 	dev_dbg(sdev->dev, "loading topology:%s\n", file);
 
-	ret = request_firmware(&fw, file, sdev->dev);
+	if (sdev->pdata->vfe) {
+		fw = &vfe_fw;
```

Comment:
Can you comment this logic

Reply:
@lgirdwood we could also use `if (sof_ops(sdev)->request_topology)` as a test here to make this usable for potential other users